src/main/webapp/scripts/popular.js

```diff
@@ -1,14 +1,20 @@
+import { getTrendingFromYoutubeApi } from './list-video-service.js';
+
 function loadPopular() {
-  loadApi(function() {
-    getTrendingFromYoutubeApi()
-    .then((response) => {
-      for(item of response.items) {
-        // Ready in a later PR: display the item nicely on the page
-        console.log(item);
-      }
-    })
-    .catch((error) => {
-      updateDom(error.message, 'popular-list-container');
-    });
+  return loadApi().then(() => popular.getTrendingFromYoutubeApi()).then((response) => {
+    for(let item of response.items) {
+      // Ready in a later PR: display the item nicely on the page
+      const containerDOM = document.getElementById('popular-list-container');
+      containerDOM.innerText += item;
+    }
+  })
+  .catch((error) => {
+    updateDom('An error occured with YouTube', 'popular-list-container');
+    console.error(error);
   });
 }
+
+let popular = { loadPopular, getTrendingFromYoutubeApi };
```

Comment:
Why is getTrendingFromYoutubeApi getting exported here? I'm guessing its for testing. Could this be exported via its origin `list-video-service` and doing the test mocks/stub by importing it in the test as well?

Reply:
getTrendingFromYoutubeApi is exported in list-video-service.js, and then both are exported here so that loadPopular can be called in testing and getTrendingFromYoutubeApi can be mocked when testing loadPopular.
If I export getTrendingFromYoutubeApi from it's original file into the test, I run into the same problem as the getRegions test where the function cannot be mocked when calling loadPopular. Modules are difficult : /
The other option is to pass getTrendingFromYoutubeApi as a parameter to loadPopular